app/multitenant/billing_emitter.go

```diff
@@ -0,0 +1,127 @@
+package multitenant
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/prometheus/client_golang/prometheus"
+	billing "github.com/weaveworks/scope/billing-client"
+	"golang.org/x/net/context"
+
+	"github.com/weaveworks/common/instrument"
+	"github.com/weaveworks/scope/app"
+	"github.com/weaveworks/scope/report"
+)
+
+var (
+	billingRequestDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "scope",
+		Name:      "billing_request_duration_seconds",
+		Help:      "Time in seconds spent emitting billing info.",
+		Buckets:   prometheus.DefBuckets,
+	}, []string{"method", "status_code"})
+)
+
+func init() {
+	prometheus.MustRegister(billingRequestDuration)
+}
+
+// BillingEmitterConfig has everything we need to make an AWS collector.
+type BillingEmitterConfig struct {
+	UserIDer        UserIDer
+	DefaultInterval time.Duration
+}
+
+type billingEmitter struct {
+	app.Collector
+
+	userIDer        UserIDer
+	defaultInterval time.Duration
+	billing         *billing.Client
+}
+
+// NewBillingEmitter is the charging-mechanism
+func NewBillingEmitter(upstream app.Collector, cfg BillingEmitterConfig) (app.Collector, error) {
+	billingClient, err := billing.New()
+	return &billingEmitter{
+		Collector:       upstream,
+		userIDer:        cfg.UserIDer,
+		defaultInterval: cfg.DefaultInterval,
```

Comment:
Personal preference is to embed the cfg struct in the component, means adding less code when adding more config.